tools/include/eosio/gen.hpp

```diff
@@ -728,19 +738,71 @@ struct generation_utils {
    inline bool is_deferred_transaction_func( const std::string& t ) {
       static const std::set<std::string> deferred_transaction_funcs =
       {
-         "eosio::internal_use_do_not_use::send_deferred",
+         "send_deferred",
       };
       return deferred_transaction_funcs.count(t) >= 1;
    }
 
    inline bool is_inline_action_func( const std::string& t ) {
       static const std::set<std::string> inline_action_funcs =
       {
-         "eosio::internal_use_do_not_use::send_inline",
-         "eosio::internal_use_do_not_use::send_context_free_inline"
+         "send_inline",
+         "send_context_free_inline"
       };
       return inline_action_funcs.count(t) >= 1;
    }
 
+   inline bool is_eosio_wasm_import_write_func( const clang::FunctionDecl *func_decl ) {
+      static const std::set<std::string> eosio_wasm_import_write_funcs =
+      {
+         "set_resource_limits",
+         "set_wasm_parameters_packed",
+         "set_resource_limit",
+         "set_proposed_producers",
+         "set_proposed_producers_ex",
+         "set_blockchain_parameters_packed",
+         "set_parameters_packed",
+         "set_kv_parameters_packed",
+         "set_privileged",
+         "db_store_i64",
+         "db_update_i64",
+         "db_remove_i64",
+         "db_idx64_store",
+         "db_idx64_update",
+         "db_idx64_remove",
+         "db_idx128_store",
+         "db_idx128_update",
+         "db_idx128_remove",
+         "db_idx256_store",
+         "db_idx256_update",
+         "db_idx256_remove",
+         "db_idx_double_store",
+         "db_idx_double_update",
+         "db_idx_double_remove",
+         "db_idx_long_double_store",
+         "db_idx_long_double_update",
+         "db_idx_long_double_remove",
+         "kv_erase",
+         "kv_set",
+         "send_deferred",
+         "send_inline",
+         "send_context_free_inline"
+      };
+
+      if (eosio_wasm_import_write_funcs.count(func_decl->getQualifiedNameAsString()) == 0) {
+         return false;
+      }
+
+      if (func_decl->isInExternCContext()) {
+         auto attrs = func_decl->getAttrs();
+         for (auto const &a : attrs) {
+            if (a->getSpelling() == "eosio_wasm_import") {
```

Comment:
This needs to be an explicit string comparison function call.  There doesn't seem to be an appropriate equality operator overload.

Reply:
Thanks. `getSpelling` returns a `const char*` but I wrongly thought it returns a std::string. It's fixed in https://github.com/EOSIO/eosio.cdt/blob/4ff779217c54dccc8fb9a59463c0e1c6cb78e82b/tools/include/eosio/gen.hpp#L799